### ISSUE TITLE
Allow feature-flags / settings to be overridden by environment

### ISSUE THREAD

**Issue by jack-pappas** [rtosholdings/riptable#269]
riptable currently only supports changing settings (e.g. number of threads to use for calculations and I/O) by calling functions of the library or setting class-level attributes.

It'd be helpful if the default values for these settings -- at least the most important ones -- could be overridden using environment variables, e.g. how numba supports changing the cache path or number of threads to be used. This would allow users to set whichever values work best for them within their Python environment and those values would be consistent across whichever riptable-using processes run under that environment (unless a user explicitly overrides the default value in their code).

One such example where this could be useful is piloting new features -- a feature-flag setting could be added to control some new feature, then a user can enable the feature in their environment for testing without having to modify any of their riptable-based code.

**Comment by rafi-sig** [comment]
I agree it would be really useful to be able to set thread limits with an environment variable or an external configuration file.
